articles/osquery-evented-tables-overview.md

```diff
@@ -0,0 +1,161 @@
+
+For the purposes of this article, we will use the term "utility" to mean the underlying OS utility that osquery subscribes to for its various evented tables. 
+
+This separation between osquery and the utility means that some evented tables rely on configurations for the utility to determine which events will be generated. At the utility level, you can specify what data is captured and at the osquery level, you can specify what information is ingested, presented, and transmitted. For most evented tables, osquery works great out-of-the-box with the utility’s default configurations,but some use cases may require adjusting the utility configuration.
+
+## How do I know whether a table is evented?
+You can tell that an osquery evented table is evented in two ways:
+
+- The table ends in `_events`.
+- In the (osquery schema)[osquery.io/schema], the table is  marked with an "evented" tag near the table name as shown here:
+
+IMAGE NEEDED HERE
+
+## What do I need to consider when configuring evented tables?
+
+### Performance impact
+Running event capture generates performance overhead from both the utility and osquery. In general, if the utility is configured loosely to generate more events, then the host is performing additional operations matching and generating the events. This further causes osquery to do more work to parse and store the extra generated events. For example, when monitoring processes, there may be frequent but low-value processes such as `awk` and `sed` which can be ignored by the utility, reducing work for the host. Collecting “good enough” data is a key method to mitigate performance impact.
+
+Furthermore, at the osquery level, the number, complexity, and frequency of queries will drive the compute resources required. Queries which utilize `WHERE` clauses will be relatively efficient (and minimize data volume), whereas many `JOIN`s or wildcards (`%`) will use more resources.
+
+Nevertheless, you may find it necessary to allocate a larger  memory footprint to osquery to satisfy your requirements. Osquery's watchdog automatically kills queries if they exceed certain system usage. Watchdog's behavior can be adjusted with the following flags:
+* `--watchdog_memory_limit` changes the maximum memory usage (expressed in MBs)
+* `--watchdog_utilization_limit` changes the maximum number of CPU cycles (defined as the `processes` table's `user_time` and `system_time`) for more than the time in seconds set by `--watchdog_latency_limit`.
+* `--watchdog_delay` sets the delay in seconds before CPU and memory usage limits will be enforced. By default, this is 60.
+
```

Comment:
```suggestion
### Performance impact

Capturing event data can be costly. The more events you are watching, the larger the load on the host. Capturing only the events you need will cut down on the amount of work osquery needs to do in order to capture and parse data.  For example, when monitoring processes, there may be frequent but low-value processes such as `awk` and `sed` which can be ignored, reducing work for the host. Collecting “good enough” data is key in managing performance impact.

Also, consider the impact of the queries you’re using to collect your event data.  Queries using `WHERE` clauses will be fairly efficient (and minimize data volume), while many `JOIN`s or wildcards (%) will use more resources. 

Even after considering all of these factors, you may need to give osquery some additional resources. Osquery's watchdog automatically kills queries if they exceed certain system usage. By default, the watchdog will allow osquery to use 200MB of memory with up to 25% CPU usage for 9 seconds. This can be adjusted with the following flags:

* `--watchdog_memory_limit` changes the maximum memory usage (expressed in MBs)
* `--watchdog_utilization_limit` changes the maximum number of CPU cycles (defined as the processes table's user_time and system_time) for more than the time in seconds set by --watchdog_latency_limit.
* `--watchdog_delay` sets the delay in seconds before CPU and memory usage limits will be enforced. By default, this is 60.

```

@zhumo, I took a quick pass at simplifying the language in the Performance impact section. I'm happy to go over the rest as well if you feel this is a value add!

Reply:
Thanks I took a bunch of your comments. Definitely the language is easier to follow. I did add back some stuff because it started to elide concepts that I think are important to keep separate.

Reply:
I'd appreciate any further sections you want to review though. It's much better after that one pass.